plugin/core/registry.py

```diff
@@ -37,13 +37,14 @@ def best_session(view: sublime.View, sessions: Iterable[Session], point: Optiona
 def get_position(view: sublime.View, event: Optional[dict] = None, point: Optional[int] = None) -> int:
     if isinstance(point, int):
         return point
-    if event is None:
-        return view.sel()[0].begin()
-    x, y = event.get("x"), event.get("y")
-    if x is not None and y is not None:
-        return view.window_to_text((x, y))
-    else:
+    if event:
+        x, y = event.get("x"), event.get("y")
+        if x is not None and y is not None:
+            return view.window_to_text((x, y))
+    try:
         return view.sel()[0].begin()
+    except IndexError:
+        return 0
```

Comment:
Didn't want to import `first_selection` in this file. I think it was frowned up.

Also, returning `0` in case of an error since can't return `None` in here. So not entirely truthful.

Reply:
Lets see this example:

Request that use `get_position` will always send a request to the server request(in worst case the position will be incorrect - 0). 
```
params = text_document_position_params(self.view, get_position(self.view, event, point))
                                                  ^^^^^^^^^^^^
session.send_request(Request(self.method, params, self.view), ...)

-----

pos = get_position(self.view, event, point)
      ^^^^^^^^^^^^
if new_name:
    return self._do_rename(pos, new_name)
```


I don't think that sending 0 as text document position params could cause an issue on the server. 

But this brings inconsistency in the LSP client. There are some requests like documentHighlight that are not send at all if there is no selection.

```python
def _do_highlights_async(self) -> None:
    region = first_selection(self.view)
    if not region:
        return  # << early return, so the request is not made.
    point = region.b
    session = self.session("documentHighlightProvider", point)
    if session:
        params = text_document_position_params(self.view, point)
        request = Request.documentHighlight(params, self.view)
        session.send_request_async(request, self._on_highlights)
```


### Question 
There are some case where LSP decided to send a request even if there is no selection(there is a fallback to 0).
But there are other cases where LSP decided not to send a request because there is no selection.

Should we make this consistent?

Reply:
changed return type to Optional